common/src/main/java/com/wynntils/features/SilencerFeature.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.mod.event.WynncraftConnectionEvent;
+import com.wynntils.core.storage.Storage;
+import com.wynntils.utils.mc.McUtils;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.sounds.SoundSource;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.UNCATEGORIZED)
+public class SilencerFeature extends Feature {
+    @RegisterConfig
+    public final Config<Double> silencerVolume = new Config<>(0.01);
+
+    @RegisterKeyBind
+    private final KeyBind silencerKeyBind =
+            new KeyBind("Toggle Silencer", GLFW.GLFW_KEY_UNKNOWN, true, this::toggleSilencer);
+
+    private final Storage<Double> lastVolume = new Storage<>(1.0);
+    private boolean isSilencerEnabled = false;
```

Comment:
```suggestion

    private boolean isSilencerEnabled = false;
```

Reply:
This boolean should be a config or storage. Please check what gammabrightfeature does and you can almost copy it as a whole, since you want to achieve the exact same behavior.

Reply:
I initially started out that way, but my understanding is that using a config or storage is primarily for reapplying the settings from the "last session". I argue this isn't the optimal approach for a feature typically activated for specific tasks, such as a grind session.

A common scenario could be players enable the Silencer, grind, and then abruptly log off when finished (ALT+F4). If the Silencer was reapplied automatically upon their return, it might lead to confusion, prompting them to manually reset the volume to a "normal" level.

With this consideration, I deemed the reapplication of settings unnecessary and therefore the use of config or storage thus I simplified the process to a regular Boolean. However, we can still use config or storage, but without reapplying the settings. Or I could make the reapplication a toggleable option but this further complicates a really simple Feature.

Id like to know what you think before I proceed with updating the code.

Reply:
Okay, I get what you mean, but I want to think about how to refactor this in a way that is both logical and somewhat signals that this is a "session only" setting.

Reply:
The problem is with reseting on quiting, how can you be so sure you correctly reset the volume when a crash/alt-f4/exit happens? I feel like this has potential of being buggy which is worse than people not remembering that it is turned on (which can be solved by an overlay like text or smth)

Reply:
In this case id suggest we instead trigger the reset onConnect or onTitleScreenInit, this way we solve both issues at the same time.

Reply:
Reset on `onTitleScreenInit`, but even that might not work with autoconnect, etc.

Reply:
Ill do some testing, i believe it will work, especially if i use the .Pre flag, as far as i remember autoconnect uses .Post

Reply:
I had missed this conversation since it was marked resolved. I think it touches on the same worry I have. There is no point in storing the lastValue unless you also store whether it is enabled or not. Possibly you should do it like this: Have two storage options, "enabled" and "volume". Whenever we enable the silencer, you set "enabled" to true and "volume" to the current volume. 

If we start Wynntils and the silencer is enabled, then we somehow exited the game with the silencer enabled (this should not really happen, so it is a crash or bug). Then you do the normal disable dance, that is, set "enabled" to false and the vanilla volume to the stored "volume".

Reply:
You can also consider merging the two stored variables into one, by setting "lastVolume" (or perhaps better `originalVolume`?) to -1 if the silencer is disabled. (Given that negative volumes are not allowed). This way you can never even get out of sync between the two variables.